code/game/objects/structures/mineral_doors.dm

```diff
@@ -83,26 +83,30 @@
 		icon_state = "[base_icon_state][smoothing_flags ? "-[smoothing_junction]" : ""]"
 
 
-/obj/structure/mineral_door/attackby(obj/item/W, mob/living/user)
+/obj/structure/mineral_door/attackby(obj/item/attacking_item, mob/living/user)
 	. = ..()
 	if(.)
 		return
 	if(QDELETED(src))
 		return
 
+	calculate_attackby(attacking_item, user)
+
+///Handles damage calculation + plasmacutter interaction from attackby
+/obj/structure/mineral_door/proc/calculate_attackby(obj/item/attacking_item, mob/living/user)
 	var/multiplier = 1
-	if(isplasmacutter(W) && !user.do_actions)
-		var/obj/item/tool/pickaxe/plasmacutter/P = W
+	if(isplasmacutter(attacking_item) && !user.do_actions)
+		var/obj/item/tool/pickaxe/plasmacutter/P = attacking_item
 		if(P.start_cut(user, src.name, src, PLASMACUTTER_BASE_COST * PLASMACUTTER_VLOW_MOD, no_string = TRUE))
 			if(istype(src, /obj/structure/mineral_door/resin))
 				multiplier += PLASMACUTTER_RESIN_MULTIPLIER //Plasma cutters are particularly good at destroying resin structures.
 			else
 				multiplier += PLASMACUTTER_RESIN_MULTIPLIER * 0.5
 			P.cut_apart(user, src.name, src, PLASMACUTTER_BASE_COST * PLASMACUTTER_VLOW_MOD) //Minimal energy cost.
-	if(W.damtype == BURN && istype(src, /obj/structure/mineral_door/resin)) //Burn damage deals extra vs resin structures (mostly welders).
+	if(attacking_item.damtype == BURN && istype(src, /obj/structure/mineral_door/resin)) //Burn damage deals extra vs resin structures (mostly welders).
 		multiplier += 1 //generally means we do double damage to resin doors
 
-	take_damage(max(0, W.force * multiplier - W.force), W.damtype, MELEE)
+	take_damage(max(0, attacking_item.force * multiplier - attacking_item.force), attacking_item.damtype, MELEE)
```

Comment:
You can early return the entire proc if its not a PC and doesn't have burn damage.